APCAppCore/APCAppCore/UI/Model/APCOnboardingManager.m

```diff
@@ -0,0 +1,238 @@
+
+@end
+
+
+@implementation APCOnboardingManager
+
+
+- (instancetype)initWithProvider:(id<APCOnboardingManagerProvider>)provider user:(APCUser * __nonnull)user {
+    if ((self = [super init])) {
+        self.provider = provider;
+        self.user = user;
+        _signInSupported = YES;
+    }
+    return self;
+}
+
++ (instancetype)managerWithProvider:(id<APCOnboardingManagerProvider>)provider user:(APCUser * __nonnull)user {
+    return [[self alloc] initWithProvider:provider user:user];
+}
+
+
+#pragma mark - Onboarding
+
+- (void)instantiateOnboardingForType:(APCOnboardingTaskType)type {
+    if (self.onboarding) {
+        self.onboarding.delegate = nil;
+        self.onboarding = nil;
+    }
+    self.onboarding = [[APCOnboarding alloc] initWithDelegate:self taskType:type];
+}
+
+- (void)userDidConsentWithResult:(ORKConsentSignatureResult *)consentResult sharingScope:(APCUserConsentSharingScope)sharingScope {
+    NSString *signatureName = [consentResult.signature.givenName stringByAppendingFormat:@" %@",consentResult.signature.familyName];
+    NSData *signatureImage = UIImagePNGRepresentation(consentResult.signature.signatureImage);
+    
+    _user.consentSignatureName = signatureName;
+    _user.consentSignatureImage = signatureImage;
+    _user.consentSignatureDate = consentResult.startDate;
+    _user.sharingScope = sharingScope;
+    _user.userConsented = YES;
+}
+
+- (void)userDeclinedConsent {
+    [[NSNotificationCenter defaultCenter] postNotificationName:APCUserDidDeclineConsentNotification object:self];
+}
+
+- (void)onboardingDidFinish {
+    [self completeOnboardingAsSignIn:(self.onboarding.taskType == kAPCOnboardingTaskTypeSignIn)];
+}
+
+- (void)onboardingDidFinishAsSignIn {
+    [self completeOnboardingAsSignIn:YES];
+}
+
+- (void)completeOnboardingAsSignIn:(BOOL)wasSignIn {
+    if (wasSignIn) {
+        self.user.signedIn = YES;
+        //[(APCAppDelegate *)[UIApplication sharedApplication].delegate afterOnBoardProcessIsFinished];     // TODO: Untwine this spaghetti
+        [[NSNotificationCenter defaultCenter] postNotificationName:APCUserSignedInNotification object:self];
+    } else {
+        self.user.signedUp = YES;
+        [[NSNotificationCenter defaultCenter] postNotificationName:APCUserSignedUpNotification object:self];
+    }
+    
+    [[NSNotificationCenter defaultCenter] postNotificationName:APCUserDidConsentNotification object:self];
+}
+
+
+#pragma mark - Permissions
+
+- (APCPermissionsManager *)permissionsManager {
+    if (!_permissionsManager) {
+        _permissionsManager = [self createPermissionsManager];
+    }
+    return _permissionsManager;
+}
+
+- (APCPermissionsManager *)createPermissionsManager {
+    APCPermissionsManager *manager = [APCPermissionsManager new];
+    
+    // set HealthKit characteristic types to read
+    NSMutableArray *types = [NSMutableArray arrayWithCapacity:3];
+    for (NSNumber *type in self.userProfileElements) {
```

Comment:
@p2 I'm debugging to investigate why HK permissions view does not pop-up in our Apple RK apps, and came across this issue. 
Here we're setting HKCharacteristicTypes - there are 3, which you cover in your switch. However, we're enumerating types which are added to the userProfileElements array by createUserProfileElements. This method adds 6 types to the array, but the switch here only covers 3 types, 2 of which are not included in userProfileElements (dateOfBirth and bloodType). It appears only biologicalSex will be added to healthKitCharacteristicTypesToRead. Did you intend to add dateOfBirth and bloodType to the createUserProfileElements method? 
Also, some apps may not require all 3, did you intend for apps to override the createUserProfileElements method by subclassing, or get types via delegation?